src/coreclr/src/zap/zapinfo.cpp

```diff
@@ -2150,15 +2150,19 @@ DWORD FilterNamedIntrinsicMethodAttribs(ZapInfo* pZapInfo, DWORD attribs, CORINF
         // is because they often change the code they emit based on what ISAs are supported by the compiler,
         // but we don't know what the target machine will support.
         //
-        // Additionally, we make sure none of the hardware intrinsic method bodies (except ARM64) get pregenerated in crossgen
+        // Additionally, we make sure none of the hardware intrinsic method bodies get pregenerated in crossgen
         // (see ZapInfo::CompileMethod) but get JITted instead. The JITted method will have the correct
         // answer for the CPU the code is running on.
 
-        // For Arm64, AdvSimd/ArmBase is the baseline that is suported and hence we do pregenerate the method bodies
-        // of ARM64 harware intrinsic.
         fTreatAsRegularMethodCall = (fIsGetIsSupportedMethod && fIsPlatformHWIntrinsic);
-#if !defined(TARGET_ARM64)
-        fTreatAsRegularMethodCall |= (!fIsPlatformHWIntrinsic && fIsHWIntrinsic);
+
+#if defined(TARGET_ARM64)
+        if (!fIsPlatformHWIntrinsic && fIsHWIntrinsic)
+        {
+            fTreatAsRegularMethodCall = (strcmp(className, "Vector64`1") != 0) && (strcmp(className, "Vector128`1") != 0);
```

Comment:
you are missing `|=`. It should be `fTreatAsRegularMethodCall |= (strcmp(className, "Vector64`1") != 0) && (strcmp(className, "Vector128`1") != 0);`

Reply:
No, I am not.

Consider this:
1) If `fTreatAsRegularMethodCall=true` before line 2160, then it means `fIsPlatformHWIntrinsic=true` and the if-condition is false.
2) If `fTreatAsRegularMethodCall=false` before line 2160, then `fTreatAsRegularMethodCall = fTreatAsRegularMethodCall || (if-cond-expr)` is equvalent to `fTreatAsRegularMethodCall = (if-cond-expr)`.

I could've written this as 
```c++
fTreatAsRegularMethodCall = (fIsGetIsSupportedMethod && fIsPlatformHWIntrinsic) || (!fIsPlatformHWIntrinsic && fIsHWIntrinsic 
#if defined(TARGET_ARM64)
    && ((strcmp(className, "Vector64`1") == 0) || (strcmp(className, "Vector128`1") == 0))
#endif
);
```

Reply:
Yes either that or something like this:
```cpp
        fTreatAsRegularMethodCall = (fIsGetIsSupportedMethod && fIsPlatformHWIntrinsic);

        if (!fIsPlatformHWIntrinsic && fIsHWIntrinsic)
        {
#if defined(TARGET_ARM64)
            // For ARM64, treat as method calls if HWIntrinsic is not the one supported.
            fTreatAsRegularMethodCall |= (strcmp(className, "Vector64`1") != 0) && (strcmp(className, "Vector128`1") != 0);
#else
            // For XArch, always treat HWIntrinsic as method calls.
  	    fTreatAsRegularMethodCall |= true;
#endif 
        }
```

Currently, it just feels like we override the `fTreatAsRegularMethodCall` without carefully inspecting the code the way you pointed.

Reply:
I pushed the simpler change. I think it's better to repeat `(!fIsPlatformHWIntrinsic && fIsHWIntrinsic)` twice.